drivers/i2c/i2c_emul.c

```diff
@@ -90,12 +90,15 @@ static int i2c_emul_transfer(const struct device *dev, struct i2c_msg *msgs, uin
 	__ASSERT_NO_MSG(emul->api);
 	__ASSERT_NO_MSG(emul->api->transfer);
 
-	ret = api->transfer(emul->target, msgs, num_msgs, addr);
-	if (ret) {
-		return ret;
+	if (emul->mock_api != NULL && emul->mock_api->transfer != NULL) {
+		ret = emul->mock_api->transfer(emul->target, msgs, num_msgs, addr);
+		printk("mock_api returned %d\n", ret);
+		if (ret != -ENOSYS) {
+			return ret;
+		}
 	}
 
-	return 0;
+	return api->transfer(emul->target, msgs, num_msgs, addr);
```

Comment:
This change looks unrelated to attributes?

Reply:
Sorry, it's a part of the tests we're using in a related change downstream. I moved one such test upstream to use this as an example (more to follow).